luadox/render/html.py

```diff
@@ -95,10 +95,15 @@ def __init__(self, parser: Parser):
         ref.flags['display'] = 'Search'
         parser.refs['--search'] = ref
 
+        # Load custom templates if provided, otherwise use default assets
+        head_template = self.config.get('project', 'head_template', fallback=None)
+        foot_template = self.config.get('project', 'foot_template', fallback=None)
+        search_template = self.config.get('project', 'search_template', fallback=None)
+
         self._templates = {
-            'head': assets.get('head.tmpl.html').decode('utf8'),
-            'foot': assets.get('foot.tmpl.html').decode('utf8'),
-            'search': assets.get('search.tmpl.html').decode('utf8'),
+            'head': open(head_template, 'r', encoding='utf8').read() if head_template else assets.get('head.tmpl.html').decode('utf8'),
+            'foot': open(foot_template, 'r', encoding='utf8').read() if foot_template else assets.get('foot.tmpl.html').decode('utf8'),
+            'search': open(search_template, 'r', encoding='utf8').read() if search_template else assets.get('search.tmpl.html').decode('utf8'),
```

Comment:
We can't assume UTF-8 here when reading the user-provided templates.  We can assume UTF-8 when they're a distributed asset since encoding is under our control, but when it's external and user-defined, it could be a different encoding.

Input encoding is configurable, so you can grab the config file for `encoding` per https://github.com/jtackaberry/luadox/blob/a5378e4f6d2db769893a9b047e085e8aeb96d7a7/luadox/main.py#L222